compiler/rustc_const_eval/src/interpret/intrinsics.rs

```diff
@@ -440,6 +455,47 @@ impl<'tcx, M: Machine<'tcx>> InterpCx<'tcx, M> {
                 self.write_scalar(Scalar::from_target_usize(align.bytes(), self), dest)?;
             }
 
+            sym::minnumf16 | sym::maxnumf16 => {
+                let a = self.read_scalar(&args[0])?.to_f16()?;
+                let b = self.read_scalar(&args[1])?.to_f16()?;
+                let res = match intrinsic_name {
+                    sym::minnumf16 => a.min(b),
+                    sym::maxnumf16 => a.max(b),
+                    _ => bug!(),
+                };
+                self.write_scalar(adjust_nan(self, res, &[a, b]), dest)?;
+            }
+            sym::minnumf32 | sym::maxnumf32 => {
+                let a = self.read_scalar(&args[0])?.to_f32()?;
+                let b = self.read_scalar(&args[1])?.to_f32()?;
+                let res = match intrinsic_name {
+                    sym::minnumf32 => a.min(b),
+                    sym::maxnumf32 => a.max(b),
+                    _ => bug!(),
+                };
+                self.write_scalar(adjust_nan(self, res, &[a, b]), dest)?;
+            }
+            sym::minnumf64 | sym::maxnumf64 => {
+                let a = self.read_scalar(&args[0])?.to_f64()?;
+                let b = self.read_scalar(&args[1])?.to_f64()?;
+                let res = match intrinsic_name {
+                    sym::minnumf64 => a.min(b),
+                    sym::maxnumf64 => a.max(b),
+                    _ => bug!(),
+                };
+                self.write_scalar(adjust_nan(self, res, &[a, b]), dest)?;
+            }
+            sym::minnumf128 | sym::maxnumf128 => {
+                let a = self.read_scalar(&args[0])?.to_f128()?;
+                let b = self.read_scalar(&args[1])?.to_f128()?;
+                let res = match intrinsic_name {
+                    sym::minnumf128 => a.min(b),
+                    sym::maxnumf128 => a.max(b),
+                    _ => bug!(),
+                };
+                self.write_scalar(adjust_nan(self, res, &[a, b]), dest)?;
+            }
```

Comment:
With just 2 types, the code duplication was bearable, but with 4 float types we really should have a single shared implementation that is generic over `rustc_apfloat::Float`. Can you try that?